crates/rattler_solve/src/libsolv/wrapper/repo.rs

```diff
@@ -1,422 +1,103 @@
-                    )
-                }
-            }
-
-            // Get the name of the package
-            if let Some((filename, package_type)) =
-                extract_known_filename_extension(&repo_data.file_name)
-            {
-                if let Some(&(other_package_type, other_solvable_id)) =
-                    package_to_type.get(filename)
-                {
-                    // A previous package that we already stored is actually a package of a better "type" so we'll just use that instead.
-                    match package_type.cmp(&other_package_type) {
-                        Ordering::Less => {
-                            unsafe {
-                                ffi::repo_free_solvable(
-                                    self.as_ptr().as_ptr(),
-                                    solvable_id.into(),
-                                    1,
-                                )
-                            };
-                            continue;
-                            // A previous package has a worse package "type", we'll reuse the handle but overwrite its attributes.
-                        }
-                        Ordering::Greater => {
-                            // Swap the "old" and "new" solvables reusing the old solvable
-                            let pool = pool.as_ref();
-                            unsafe {
-                                let solvables = std::slice::from_raw_parts_mut(
-                                    pool.solvables,
-                                    pool.nsolvables as _,
-                                );
-                                solvables.swap(solvable_id.0 as _, other_solvable_id.0 as _);
-                                ffi::repodata_swap_attrs(
-                                    data,
-                                    solvable_id.into(),
-                                    other_solvable_id.into(),
-                                );
-                                ffi::repo_free_solvable(
-                                    self.as_ptr().as_ptr(),
-                                    solvable_id.into(),
-                                    1,
-                                );
-                            }
-                            package_to_type.insert(filename, (package_type, other_solvable_id));
-                        }
-                        Ordering::Equal => {
-                            // They both have the same extension? Keep them both I guess?
-                            unimplemented!("found a duplicate package")
-                        }
-                    }
-                } else {
-                    package_to_type.insert(filename, (package_type, solvable_id));
-                };
-            } else {
-                tracing::warn!("unknown package extension: {}", &repo_data.file_name);
-            }
+    /// Adds a new repodata to this repo
```

Comment:
Maybe explain a little bit here what repodata is, or at least that its a libsolv datastructure.

Reply:
Will do